app/views/CreateChannelView.tsx

```diff
@@ -73,357 +74,295 @@ interface IOtherUser {
+			encryptionEnabled: state.encryption.enabled,
+			users: state.selectedUsers.users,
+			useRealName: state.settings.UI_Use_Real_Name as boolean
+		}),
+		shallowEqual
+	);
+
+	const permissions = usePermissions(['create-c', 'create-p']);
+
+	useLayoutEffect(() => {
 		navigation.setOptions({
 			title: isTeam ? I18n.t('Create_Team') : I18n.t('Create_Channel')
 		});
-	};
-
-	toggleRightButton = (channelName: string) => {
-		const { navigation } = this.props;
-		navigation.setOptions({
-			headerRight: () =>
-				channelName.trim().length > 0 && (
-					<HeaderButton.Container>
-						<HeaderButton.Item title={I18n.t('Create')} onPress={this.submit} testID='create-channel-submit' />
-					</HeaderButton.Container>
-				)
-		});
-	};
-
-	onChangeText = (channelName: string) => {
-		this.toggleRightButton(channelName);
-		this.setState({ channelName });
-	};
-
-	submit = () => {
-		const { channelName, type, readOnly, broadcast, encrypted, isTeam } = this.state;
-		const { users: usersProps, isFetching, dispatch } = this.props;
+	}, [isTeam, navigation]);
 
+	const submit = () => {
 		if (!channelName.trim() || isFetching) {
 			return;
 		}
 
 		// transform users object into array of usernames
-		const users = usersProps.map(user => user.name);
+		const usersMapped = users.map(user => user.name);
 
 		// create channel or team
 		const data = {
 			name: channelName,
-			users,
+			users: usersMapped,
 			type,
 			readOnly,
 			broadcast,
 			encrypted,
 			isTeam,
-			teamId: this.teamId
+			teamId
 		};
 		dispatch(createChannelRequest(data));
 		Review.pushPositiveEvent();
 	};
 
-	removeUser = (user: IOtherUser) => {
-		logEvent(events.CR_REMOVE_USER);
-		const { dispatch } = this.props;
-		dispatch(removeUser(user));
-	};
+	const onValueChangeType = useCallback(
+		(value: boolean) => {
+			logEvent(events.CR_TOGGLE_TYPE);
+			// If we set the channel as public, encrypted status should be false
+			setType(value);
+			setEncrypted(value && encrypted);
+		},
+		[encrypted]
+	);
+
+	const renderType = () => {
+		const isDisabled = permissions.filter(r => r === true).length <= 1;
+
+		let hint = '';
+		if (isTeam && type) {
+			hint = 'Team_hint_private';
+		}
+		if (isTeam && !type) {
+			hint = 'Team_hint_public';
+		}
+		if (!isTeam && type) {
+			hint = 'Channel_hint_private';
+		}
+		if (!isTeam && !type) {
+			hint = 'Channel_hint_public';
+		}
 
-	renderSwitch = ({ id, value, label, onValueChange, disabled = false }: ISwitch) => {
-		const { theme } = this.props;
 		return (
-			<View style={[styles.switchContainer, { backgroundColor: themes[theme].backgroundColor }]}>
-				<Text style={[styles.label, { color: themes[theme].titleText }]}>{I18n.t(label)}</Text>
-				<Switch
-					value={value}
-					onValueChange={onValueChange}
-					testID={`create-channel-${id}`}
-					trackColor={SWITCH_TRACK_COLOR}
-					disabled={disabled}
-				/>
-			</View>
+			<RenderSwitch
+				id={'type'}
+				value={permissions[1] ? type : false}
```

Comment:
We agreed not to use `permissions[N]` when we implemented `usePermissions`.
Instead you should make it clear on your call.
```tsx
const [nameYourPermission, nameIt] = usePermissions(['create-c', 'create-p']);
```